tests/foreman/ui/test_audit.py

```diff
@@ -0,0 +1,258 @@
+# -*- encoding: utf-8 -*-
+"""Test class for Audit UI
+
+:Requirement: Audit
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+from nailgun import entities
+from robottelo.decorators import run_in_one_thread, run_only_on, tier1
+from robottelo.test import UITestCase
+from robottelo.ui.factory import make_role
+from robottelo.ui.session import Session
+
+
+@run_in_one_thread
+class AuditTestCase(UITestCase):
+    """Implements Audit tests from UI"""
+
+    @run_only_on('sat')
+    @tier1
+    def test_positive_create_by_type(self):
+        """Create entities of different types and check audit logs for these
+        events using entity type and performed action as search criteria
+
+        :id: 26197b39-4d56-4aab-8df8-f0fcedbffdb7
+
+        :expectedresults: Audit logs contain corresponding entries per each
+            create event
+
+        :CaseImportance: Critical
+        """
+        with Session(self):
+            for entity_item in [
+                {
+                    'entity': entities.Architecture(),
+                    'entity_type': 'architecture'
+                },
+                {
+                    'entity': entities.AuthSourceLDAP(),
+                    'entity_type': 'auth_source',
+                    'value_template': 'LDAP-{entity.name}'
+                },
+                {
+                    'entity': entities.ComputeProfile(),
+                    'entity_type': 'compute_profile'
+                },
+                {
+                    'entity': entities.LibvirtComputeResource(),
+                    'entity_type': 'compute_resource',
+                    'value_template': '{entity.name} (Libvirt)'
+                },
+                {
+                    'entity': entities.ConfigGroup(),
+                    'entity_type': 'config_group'
+                },
+                {'entity': entities.Domain(), 'entity_type': 'domain'},
+                {'entity': entities.Host(), 'entity_type': 'host'},
+                {'entity': entities.HostGroup(), 'entity_type': 'hostgroup'},
+                {'entity': entities.Image(
```

Comment:
Imo you shouldn't initialize entities at this stage and store them altogether, it's a list of data to work on later. Similar as `.create()` - you create a single entity inside for block, not in its definition. Dependent sub-entities aren't an excuse here, as you can use some `**entity_item['entity_kwargs']` to unpack them.

Reply:
Thinking of it, it's not a big deal but rather suggestion for optimization, so maybe disregard, it's definitely not a blocker.